third_party/amd/lib/Dialect/TritonAMDGPU/IR/Dialect.cpp

```diff
@@ -132,6 +129,46 @@ LogicalResult ExtractSliceOp::verify() {
                        << shapePerCTATile << "]";
   }
 
+  auto llSrc = triton::gpu::toLinearLayout(srcShape, srcLayout);
+  auto llRes = triton::gpu::toLinearLayout(resShape, resultLayout);
+  auto ctx = srcTy.getContext();
+
+  auto kReg = mlir::StringAttr::get(ctx, "register");
+  auto kLane = mlir::StringAttr::get(ctx, "lane");
+  auto kWarp = mlir::StringAttr::get(ctx, "warp");
+  auto kBlock = mlir::StringAttr::get(ctx, "block");
+
+  const auto &srcBases = llSrc.getBases();
+  const auto &regsSrc = srcBases.find(kReg)->second;
+  const auto &lanesSrc = srcBases.find(kLane)->second;
+  const auto &warpsSrc = srcBases.find(kWarp)->second;
+  const auto &blocksSrc = srcBases.find(kBlock)->second;
+
+  const auto &resBases = llRes.getBases();
+  const auto &regsRes = resBases.find(kReg)->second;
+  const auto &lanesRes = resBases.find(kLane)->second;
+  const auto &warpsRes = resBases.find(kWarp)->second;
+  const auto &blocksRes = resBases.find(kBlock)->second;
+
+  if (blocksSrc != blocksRes || lanesSrc != lanesRes || warpsSrc != warpsRes ||
+      blocksSrc != blocksRes) {
+    return emitError(
+        "result LL must match source LL lane, warp and block basis");
+  }
+
+  llvm::outs() << regsRes.size() << " " << regsSrc.size() << "\n";
+  if (regsRes.size() > regsSrc.size()) {
+    return emitError("result LL register dim size must me smaller than source "
+                     "LL register dim");
+  }
+
+  for (int i = 0; i < regsRes.size(); i++) {
+    if (regsRes[i] != regsRes[i]) {
+      return emitError("result LL register bases must be subset of source LL "
+                       "register bases");
+    }
+  }
+
```

Comment:
Should we also check that the number of dropped register basis vector is consistent with the number of slices? E.g., if we drop 3 basis vectors we should slice into 8 slices?